cli/src/pcluster/models/s3_bucket.py

```diff
@@ -148,12 +148,96 @@ def configure_s3_bucket(self):
             bucket_name=self.name,
             configuration={"Rules": [{"ApplyServerSideEncryptionByDefault": {"SSEAlgorithm": "AES256"}}]},
         )
-        deny_http_policy = (
-            '{{"Id":"DenyHTTP","Version":"2012-10-17","Statement":[{{"Sid":"AllowSSLRequestsOnly","Action":"s3:*",'
-            '"Effect":"Deny","Resource":["arn:{partition}:s3:::{bucket_name}","arn:{partition}:s3:::{bucket_name}/*"],'
-            '"Condition":{{"Bool":{{"aws:SecureTransport":"false"}}}},"Principal":"*"}}]}}'
-        ).format(bucket_name=self.name, partition=self.partition)
-        AWSApi.instance().s3.put_bucket_policy(bucket_name=self.name, policy=deny_http_policy)
+
+        bucket_policy = self._generate_bucket_policy()
+        AWSApi.instance().s3.put_bucket_policy(bucket_name=self.name, policy=json.dumps(bucket_policy))
+
+    def _generate_bucket_policy(self):
```

Comment:
We should cover the generation of this bucket policy with a unit test. I know this policy exist before your PR, but it's worth it to take the chance to cover it.

Reply:
Done